src/shared/mount-util.c

```diff
@@ -1138,7 +1138,7 @@ static int mount_in_namespace(
         _cleanup_close_pair_ int errno_pipe_fd[2] = EBADF_PAIR;
         pid_t child;
 
-        if (flags & MOUNT_IN_NAMESPACE_IS_IMAGE) {
+        if (!(flags & MOUNT_IN_NAMESPACE_UMOUNT) && flags & MOUNT_IN_NAMESPACE_IS_IMAGE) {
```

Comment:
this conditionalization is not sufficient.

i.e. the chase() on the source above should not be done if you don't actually care about the source.

Reply:
also, more importantly, this leaves the codepath where mount_new_api_supported() says no out in the cold. which needs to be covered.

Reply:
i am kinda puzzled this passed CI, I assumed that we still have kernels in our CI that do not support the new mount api...

anyway, can you please add an explicit test case for umount-in-namespace covering this?

Reply:
The added test should already cover this: https://github.com/systemd/systemd/pull/33995/commits/1e4e23cb7e75a0d545633251effb8dbfc373079c#diff-d759d818ef340a3b3af79f45d89215ac60bef17b535061775d67a178cbc44955R577

I guess we don't have that old kernels anymore? Do we even need to bother with them for _new_ things? Honestly I'd just make it a requirement for this new feature and check if we have the new mount api in service_should_reload_extensions() and return false otherwise. This is all new stuff anyway.

Reply:
i don't really see why umounting a dir in a namespace would require new mount api though...

i think the right approach here is to uncouple mounting from unmounting here, and the issue just goes away.